packages/aws-lambda/src/triggers.js

```diff
@@ -212,43 +212,112 @@ function extractEventFromSQS(event, span) {
 }
 
 function isInvokeFunction(context) {
-  const custom =
-    (context && context.clientContext && context.clientContext.Custom) ||
-    (context && context.clientContext && context.clientContext.custom);
+  const custom = readClientContextCustom(context);
   return custom && (custom['x-instana-l'] || custom['x-instana-s'] || custom['x-instana-t']);
 }
 
 exports.readTracingHeaders = function readTracingHeaders(event, context) {
   const tracingHeaders = {};
-  // Node.js AWS SDK documentation expects for Custom, with capital "C", but the same is not explicitly said for
-  // other languages, so we test both
-  const custom =
-    (context && context.clientContext && context.clientContext.Custom) ||
-    (context && context.clientContext && context.clientContext.custom);
+
   if (event.headers && typeof event.headers === 'object') {
-    let lowerCaseKey;
-    Object.keys(event.headers).forEach(key => {
-      if (typeof key === 'string') {
-        lowerCaseKey = key.toLowerCase();
-        if (lowerCaseKey === tracingConstants.traceIdHeaderNameLowerCase) {
-          tracingHeaders.t = event.headers[key];
-        } else if (lowerCaseKey === tracingConstants.spanIdHeaderNameLowerCase) {
-          tracingHeaders.s = event.headers[key];
-        } else if (lowerCaseKey === tracingConstants.traceLevelHeaderNameLowerCase) {
-          tracingHeaders.l = event.headers[key];
-        }
-      }
-    });
-  } else if (custom) {
-    if (custom['x-instana-l']) {
-      tracingHeaders.l = custom['x-instana-l'];
+    tracingHeaders.t = tracingUtil.readAttribCaseInsensitive(event.headers, tracingConstants.traceIdHeaderName);
+    tracingHeaders.s = tracingUtil.readAttribCaseInsensitive(event.headers, tracingConstants.spanIdHeaderName);
+    tracingHeaders.l = tracingUtil.readAttribCaseInsensitive(event.headers, tracingConstants.traceLevelHeaderName);
+    if (hasTracingHeaders(tracingHeaders)) {
+      return tracingHeaders;
+    }
+  }
+
+  const custom = readClientContextCustom(context);
+  if (custom) {
+    if (custom[tracingConstants.traceLevelHeaderNameLowerCase]) {
+      tracingHeaders.l = custom[tracingConstants.traceLevelHeaderNameLowerCase];
     }
-    if (custom['x-instana-s']) {
-      tracingHeaders.s = custom['x-instana-s'];
+    if (custom[tracingConstants.spanIdHeaderNameLowerCase]) {
+      tracingHeaders.s = custom[tracingConstants.spanIdHeaderNameLowerCase];
     }
-    if (custom['x-instana-t']) {
-      tracingHeaders.t = custom['x-instana-t'];
+    if (custom[tracingConstants.traceIdHeaderNameLowerCase]) {
+      tracingHeaders.t = custom[tracingConstants.traceIdHeaderNameLowerCase];
+    }
+    if (hasTracingHeaders(tracingHeaders)) {
+      return tracingHeaders;
+    }
+  }
+
+  if (isSQSTrigger(event)) {
+    const sqsMessageAttributes = event.Records[0].messageAttributes;
```

Comment:
Yes, that is guaranteed, because `isSQSTrigger` checks for that:

```
return Array.isArray(event.Records) && event.Records.length > 0 && event.Records[0].eventSource === 'aws:sqs';
```